packages/rehype-plugin-codeblock/src/metastring.ts

```diff
@@ -55,10 +58,98 @@ export function parseMetastring(
     );
   }
 
-  // Parse line highlights.
-  // TODO(imballinst): not a priority now since it's more recommended to use inline ones.
+  // Parse highlight lines.
+  // The format is as the following:
+  //
+  // {1}, or {1,3}, or {1,3,5-9}
+  //
+  // For the segments with the dash, it means range.
+  // Hence, 5-9 means 5,6,7,8,9.
+  const highlightedLines: number[] = [];
+  const lineHighlightStartIndex = metastring.indexOf('{');
+
+  if (lineHighlightStartIndex > -1) {
+    const lineHighlightEndIndex = metastring.indexOf(
+      '}',
+      lineHighlightStartIndex
+    );
+    const segments = metastring
+      .slice(lineHighlightStartIndex + 1, lineHighlightEndIndex)
+      .split(',');
+
+    for (const segment of segments) {
+      if (segment.includes('-')) {
+        // "Expand" ranges.
+        const segmentArray = segment.split('-');
+        const start = Number(segmentArray[0]);
+        const end = Number(segmentArray[1]);
+
+        // Push all number from the range.
+        for (let i = start; i <= end; i++) {
+          // It's -1 because we want to address array indices.
+          highlightedLines.push(i - 1);
+        }
+      } else {
+        // Push one number.
+        // It's -1 because we want to address array indices.
+        highlightedLines.push(Number(segment) - 1);
+      }
+    }
+  }
+
   return {
     title,
-    isCollapsible
+    isCollapsible,
+    highlightedLines
   };
 }
+
+export function stripTitleFromElementProperties(
```

Comment:
This is the same as the other PR right?

Reply:
Yes, sorry, this PR uses the other PR so that it's easier to resolve the conflicts. The changes should be cleaner now: https://github.com/imballinst/protosaurus/pull/35/files.